pynest/examples/pynest-microcircuit/network.py

```diff
@@ -0,0 +1,347 @@
+                    if j % 2:
+                        syn_dict_rec = syn_dict_rec_in
+                    else:
+                        syn_dict_rec = syn_dict_rec_ex
+                    nest.Connect(source_pop, target_pop,
+                                 conn_spec=conn_dict_rec,
+                                 syn_spec=syn_dict_rec)
+        print 'Recurrent connections established'
+
+    # poisson generators connected to the different populations
+    def connect_poisson(self):
+        for i, target_pop in enumerate(self.pops):
+                conn_dict_poisson = {'rule': 'all_to_all'}
+                syn_dict_poisson = {'model': 'static_synapse',
+                                    'weight': self.w_ext,
+                                    'delay': self.net_dict['poisson_delay']}
+                nest.Connect(self.poisson[i], target_pop,
+                             conn_spec=conn_dict_poisson,
+                             syn_spec=syn_dict_poisson)
+        print 'Poisson background input connected'
+
+    # the thalamic neurons are connected to the different populations
+    def connect_thalamus(self):
+        for i, target_pop in enumerate(self.pops):
+                conn_dict_th = {'rule': 'fixed_total_number',
+                                'N': int(self.nr_synapses_th[i])}
+                syn_dict_th = {'weight': {'distribution': 'normal_clipped',
+                                          'mu': self.thalamic_weight,
+                                          'sigma': (self.thalamic_weight *
+                                                    self.net_dict['PSP_sd']),
+                                          'low': 0.0},
+                               'delay': {'distribution': 'normal_clipped',
+                                         'mu': self.stim_dict['delay_th'][i],
+                                         'sigma': stim_dict['delay_th_sd'][i],
+                                         'low': self.sim_resolution}}
+                nest.Connect(self.thalamic_population, target_pop,
+                             conn_spec=conn_dict_th, syn_spec=syn_dict_th)
+        print 'Thalamus connected'
+
+    # the dc generators are connected to the neuronal populations
+    def connect_dc_generator(self):
+        for i, target_pop in enumerate(self.pops):
+            if self.stim_dict['dc_input']:
+                nest.Connect(self.dc[i], target_pop)
+        print 'DC Generator connected'
+
+    # the recording devices are connected to the neuronal populations
+    def connect_devices(self):
+        for i, target_pop in enumerate(self.pops):
+            if 'voltmeter' in self.net_dict['rec_dev']:
+                nest.Connect(self.voltmeter[i], target_pop)
+            if 'spike_detector' in self.net_dict['rec_dev']:
+                nest.Connect(target_pop, self.spikedetector[i])
+        print '%s of 2 Devices connected' % (len(self.net_dict['rec_dev']))
+
+    def connect(self):
```

Comment:
Since this function sets up the entire network (incl. creating the populations), please consider renaming it to `setup`.